lib/AST/ASTContext.cpp

```diff
@@ -908,6 +914,75 @@ FuncDecl *ASTContext::getEqualIntDecl() const {
   return nullptr;
 }
 
+FuncDecl *ASTContext::getMutatingXorIntDecl() const {
```

Comment:
I'll tinker with factoring the commonalities out into a function that takes a callback and loop back around.